xds/internal/client/client.go

```diff
@@ -0,0 +1,47 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package client contains the implementation of the xds client used by xds.
+package client
+
+import "context"
+
+// ServiceUpdate contains update about the service.
+type ServiceUpdate struct{}
+
+// ClusterUpdate contains update about the cluster.
+type ClusterUpdate struct{}
+
+// EndpointUpdate contains update about the endpoint.
+type EndpointUpdate struct{}
+
+// Client is the xds client resolver and balancer use to watch xds updates.
+//
+// All the watch methods don't block.
+type Client interface {
```

Comment:
Also, should this be an exported interface if the only reason for having the interface is to be able to mock out in tests.

Reply:
Yeah, the main reason for the exported type to be an interface is for testing.
Or, just exporting the real `Client` sounds OK, too, if that looks better eventually.
We can still mock it in tests like
https://github.com/grpc/grpc-go/blob/7c3115d8bba8df106f26e55d52db456ed42893f8/xds/internal/balancer/xds.go#L99-L104

Reply:
Made it a struct instead.